drivers/sensor/adxl345/adxl345.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2020 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/sensor.h>
+#include <init.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <logging/log.h>
+
+#include "adxl345.h"
+
+LOG_MODULE_REGISTER(ADXL345, CONFIG_SENSOR_LOG_LEVEL);
+
+static int adxl345_read_reg(struct device *dev, u8_t reg)
+{
+	struct adxl345_dev_data *data = dev->driver_data;
+	u8_t reg_val;
+	int rc = i2c_write_read(data->i2c_master,
+				data->i2c_addr,
+				&reg, 1,
+				&reg_val, 1);
+
+	if (rc < 0) {
+		LOG_ERR("Register read failed with rc=%d\n", rc);
+		return rc;
+	}
+
+	return reg_val;
+}
+
+static int adxl345_write_reg(struct device *dev, u8_t reg, u8_t val)
+{
+	struct adxl345_dev_data *data = dev->driver_data;
+	int rc;
+	u8_t write_buf[2];
+
+	write_buf[0] = reg;
+	write_buf[1] = val;
+	rc = i2c_write(data->i2c_master,
+		       write_buf, 2,
+		       data->i2c_addr);
+	if (rc < 0) {
+		LOG_ERR("Writing register failed\n");
+		return rc;
+	}
+
+	return 0;
+}
+
+static int adxl345_read_sample(struct device *dev,
+			       struct adxl345_sample *sample)
+{
+	struct adxl345_dev_data *data = dev->driver_data;
+	s16_t raw_x, raw_y, raw_z;
+	u8_t axis_data[6];
+
+	int rc = i2c_burst_read(data->i2c_master,
+				data->i2c_addr,
+				ADXL345_X_AXIS_DATA_0_REG,
+				axis_data,
+				6);
+
+	if (rc < 0) {
+		LOG_ERR("Samples read failed with rc=%d\n", rc);
+		return rc;
+	}
+
+	raw_x = axis_data[0] | (axis_data[1] << 8);
+	raw_y = axis_data[2] | (axis_data[3] << 8);
+	raw_z = axis_data[4] | (axis_data[5] << 8);
+
+	if (raw_x & (1 << 9)) {
```

Comment:
Please replace `(1 << 9)` with `BIT(9)` for all three axes

Reply:
Done